cpp/src/arrow/compute/logical_type.h

```diff
@@ -73,6 +73,8 @@ class ARROW_EXPORT LogicalType {
 
   Id id() const { return id_; }
 
+  virtual ~LogicalType() = 0;
```

Comment:
This should probably be `default`. I will handle in my revisions for code review comments

Reply:
Need to implement it too: `LogicalType::~LogicalType() {}`

Reply:
fixed